src/hotspot/share/utilities/events.cpp

```diff
@@ -163,8 +165,11 @@ void ExceptionsEventLog::log(Thread* thread, Handle h_exception, const char* mes
                   _records[index].data.size());
   st.print("Exception <");
   h_exception->print_value_on(&st);
-  st.print("%s%s> (" PTR_FORMAT ") \n"
+  if (message != nullptr) {
+    int len = message_length_limit > 0 ? message_length_limit : (int)strlen(message);
```

Comment:
Do we need to check that `message_length_limit <= (int)strlen(message)`?
Or it is intentional to reserve bigger space in output than `message`?

Reply:
I don't quite understand the question. We are specifying the maximum number of characters to print: it is either `message_length_limit` if that is > 0 otherwise it is no limit i.e. the whole string i.e. `strlen(message)`.

Reply:
Someone can pass `message_length_limit` which is > strlen(message). What happens in this case?

Reply:
Printing stops at the end of `message`. Given this:
```
const char* msg = "This is the message";
printf(">>%.*s<<\n", 40, msg);
```
we get:
```
>>This is the message<<
```
Having a limit >> strlen(message) is the normal expected case.

Reply:
Good.